articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,161 @@
+This is the part where one would add the DNS records in their DNS management dashboard. If you are running Nginx locally, you can skip this step.
+
+The general DNS Configurations would be something like
+- Server app mapped to the server.domain
+- Client app mapped to the client.domain
+
+My Localhost Config, in this case, would be
+- Server mapped to server.localhost
+- Client mapped to client.localhost
+
+There are two standard protocols HTTP and HTTPS. The default port for HTTP is 80 and HTTPS is 443. This is the reason we must not run our applications on these ports because our NGINX server would be running on these two ports. All the requests the client makes would either be redirected to Port 80 or 443 from where it would be redirected internally to the corresponding application.
+
+## Step 4 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX Reverse Proxy to make it all work.
+Navigate into the Nginx folder and locate  ```nginx.conf``` .
+
+Add these configurations inside the HTTP block.
+
+### Step 4.1 - HTTP
+
+```
+server {       
+     listen       80;
+     server_name  server.domain;
+     location / {
+          proxy_pass "http://localhost:3000" ;
+     }
+}
+server {       
+     listen       80;
+     server_name  client.domain;
+
+     location / {
+          proxy_pass "http://localhost:3001" ;
+     }
+}
+```
+
+### Step 4.2 - HTTPS
+
+```
+server{
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name server.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3000/";
+    }
+
+    ssl_certificate <location of SSL certificate>
+    ssl_certificate_key <location of SSL certificate Key>
+}
+server{
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name client.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3001/";
+    }
+
+    ssl_certificate <location of SSL certificate>
+    ssl_certificate_key <location of SSL certificate Key>
+}
+```
+
+**NOTE** 
+These are the basic minimum configurations required to successfully configure NGINX for reverse proxying. Feel free to explore other config parameters as well. 
+
+Change the domain name to your domain.
+For **a** SSL Certificate and Key, you can **obtain them** from your SSL provider. If you don't have one, Use this free service [LetsEncrypt](https://letsencrypt.org/). Follow their documentation to get Free SSL instantly!
+
+
+
+## Step 5 - Save and Restart
+
+After editing, save your changes. Use the ```nginx -t``` command to test your changes before actually reloading NGINX. It is good practice do that to make sure your server doesn't crash if there were any errors in your config file. Once you get a message that the test is successful, you can go ahead and restart NGINX.
+Open the browser and enter the URLs to find your applications running on the corresponding URLs configured.
```

Comment:
Need to add an extra line break before Open.

Reply:
Missed that, got it now. Thanks